P3 metrics (#1137)
Macrometacorp/docs: docs/photoniq/p3/p3-metrics.md

---
sidebar_position: 30
title: P3 Metrics
---

> PhotonIQ Performance Proxy (P3) offers a comprehensive suite of metrics designed to provide insights into both the performance improvements and operational status of your web optimization efforts. These metrics help you understand how effectively P3 enhances your website's speed, reliability, and user experience.

Reviewer: ```suggestion
PhotonIQ Performance Proxy (P3) offers a comprehensive suite of metrics designed to provide insights into no of requests coming in, no of optimized pages and no of optimized requests. It also provides an insight into how much time it takes for P3 to analyze and validate the optimizations before applying to it. It also tells us that how requests have failed and how many are successful . 
```

These metrics collectively give you a detailed view of the operational efficiency, performance impact, and optimization success of P3, enabling technical teams to make informed decisions and continuously refine their web performance strategies. This page provides a detailed list of key metrics provided by P3.

## Requests Metrics

- **Requests Count (1hr, 1d, 1m)**: The total number of requests processed by P3 over the past hour, day, and month, respectively.

> - **Request Errors Count (1hr, 1d, 1m)**: The number of requests that resulted in errors during processing over the past hour, day, and month.

Reviewer: ```suggestion
- **Request Errors Count (1hr, 1d, 1m)**: The number of requests that resulted in errors when requesting the content from the origin the past hour, day, and month.
```

> - **Requests Concurrency (1m, 5m)**: Measures the number of concurrent requests handled by P3 within 1-minute and 5-minute intervals, indicating the load and traffic handling capability.

Reviewer: remove this for now

> - **Requests Optimized Count (1hr, 1d, 1m)**: Counts the requests that were successfully optimized by P3 in the last hour, day, and month, showcasing the efficiency of optimizations.

Reviewer: ```suggestion
- **Requests Optimized Count (1hr, 1d, 1m)**: Counts the requests that were successfully optimized by P3 in the last hour, day, and month.
```

> - **Requests Unoptimized Count (1hr, 1d, 1m)**: The count of requests that were not optimized over the past hour, day, and month, providing insights into potential areas for improvement.

Reviewer: ```suggestion
- **Requests Unoptimized Count (1hr, 1d, 1m)**: The count of requests that were not optimized over the past hour, day, and month, providing insights into potential areas for improvement.
```

## Optimizer Metrics

> - **Policy Count**: The total number of optimization policies currently active within P3, reflecting the customization and strategy applied to content optimization.

Reviewer: Remove this

- **Pages Optimized**: Total pages that have been successfully optimized, indicating the scope and impact of P3's optimization efforts.
- **Pages Failed to Optimize**: The number of pages that could not be optimized, highlighting challenges or limitations faced during the optimization process.
- **Average Optimization Time (seconds)**: The average time taken to optimize a page, measured in seconds. This metric helps in evaluating the efficiency of the optimization process.

> - **Average Validation Time (seconds)**: Average time spent validating the optimizations applied to ensure they meet the desired standards and do not negatively impact the user experience.

Reviewer: ```suggestion
- **Average Validation Time (seconds)**: Average time spent validating the optimizations applied to ensure the page doesn't visually break and to ensure the optimizations doesnt break the functionality of the page 
```

> - **Average Analyzer Time (seconds)**: Reflects the average time taken by the Offline Analyzer to assess and formulate optimization strategies for a webpage.

Reviewer: ```suggestion
- **Average Analyzer Time (seconds)**: Reflects the average time taken by the Offline Analyzer to assess and formulate optimization strategies for the incoming URL which matches the configured URL pattern in policy
```

> - **Average Queue Length**: Measures the average number of optimization tasks waiting to be processed, offering insights into the workload and operational throughput of P3.

Reviewer: remove thsi

> - **Average Queue Length**: Measures the average number of optimization tasks waiting to be processed, offering insights into the workload and operational throughput of P3.

Reviewer: Add these 
passthrough pages--- no of pages that are not configured to use P3 but goes through P3. Can give us insight that other pages if optimized can improve the performance of the page
passthrough requests---> no of requests  that are not configured to use P3 but goes through P3. Can give us insight that other pages if optimized can improve the performance of the page.
Difference between passthrough pages and passthrough requests is that pages will be unique and in requests same page can be counted multiple times if the request for that page comes to p3 multiple time
optimized pages--->no of pages that are optimized  optimized by P3
optimized requests--->no of requests that are optimized  optimized by P3
unoptimized pages-- no of pages that are not optimized or cant be optimized at the moment when the offline analyzer is doing its thing 
unoptimized requests-->no of requests that are not optimized or cant be optimized at the moment when the offline analyzer is doing its thing .
